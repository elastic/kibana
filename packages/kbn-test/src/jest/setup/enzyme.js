/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
