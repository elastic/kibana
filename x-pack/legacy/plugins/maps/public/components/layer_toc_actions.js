/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React, { Component, Fragment } from 'react';

import {
  EuiButtonEmpty,
  EuiPopover,
  EuiContextMenu,
  EuiIcon,
  EuiToolTip
} from '@elastic/eui';
import { i18n } from '@kbn/i18n';

export class LayerTocActions extends Component {

  state = {
    isPopoverOpen: false,
    supportsFitToBounds: false,
  };

  componentDidMount() {
    this._isMounted = true;
    this._loadSupportsFitToBounds();
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  async _loadSupportsFitToBounds() {
    const supportsFitToBounds = await this.props.layer.supportsFitToBounds();
    if (this._isMounted) {
      this.setState({ supportsFitToBounds });
    }
  }

  _togglePopover = () => {
    this.setState(prevState => ({
      isPopoverOpen: !prevState.isPopoverOpen,
    }));
  };

  _closePopover = () => {
    this.setState(() => ({
      isPopoverOpen: false
    }));
  };

  _renderPopoverToggleButton() {
    const {
      icon,
      tooltipContent,
      footnotes,
    } = this.props.layer.getIconAndTooltipContent(this.props.zoom, this.props.isUsingSearch);

    const footnoteIcons = footnotes.map((footnote, index) => {
      return <Fragment key={index}>{''}{footnote.icon}</Fragment>;
    });
    const footnoteTooltipContent = footnotes.map((footnote, index) => {
      return <div key={index}>{footnote.icon}{' '}{footnote.message}</div>;
    });

    return (
      <EuiToolTip
        anchorClassName="mapLayTocActions__tooltipAnchor"
        position="top"
        title={this.props.displayName}
        content={
          <Fragment>
            {tooltipContent}
            {footnoteTooltipContent}
          </Fragment>
        }
      >
        <EuiButtonEmpty
          className="mapTocEntry__layerName eui-textLeft"
          size="xs"
          flush="left"
          color="text"
          onClick={this._togglePopover}
          data-test-subj={`layerTocActionsPanelToggleButton${this.props.escapedDisplayName}`}
        >
          <span className="mapTocEntry__layerNameIcon">{icon}</span>
          {this.props.displayName}
          {' '}
          {footnoteIcons}
        </EuiButtonEmpty>
      </EuiToolTip>
    );
  }

  _getActionsPanel() {
    const actionItems = [
      {
        name: i18n.translate('xpack.maps.layerTocActions.fitToDataTitle', {
          defaultMessage: 'Fit to data',
        }),
        icon: (
          <EuiIcon
            type="search"
            size="m"
          />
        ),
        'data-test-subj': 'fitToBoundsButton',
        toolTipContent: this.state.supportsFitToBounds ? null : i18n.translate('xpack.maps.layerTocActions.noFitSupportTooltip', {
          defaultMessage: 'Layer does not support fit to data',
        }),
        disabled: !this.state.supportsFitToBounds,
        onClick: () => {
          this._closePopover();
          this.props.fitToBounds();
        },
      },
      {
        name: this.props.layer.isVisible() ? i18n.translate('xpack.maps.layerTocActions.hideLayerTitle', {
          defaultMessage: 'Hide layer',
        }) : i18n.translate('xpack.maps.layerTocActions.showLayerTitle', {
          defaultMessage: 'Show layer',
        }),
        icon: (
          <EuiIcon
            type={this.props.layer.isVisible() ? 'eye' : 'eyeClosed'}
            size="m"
          />
        ),
        'data-test-subj': 'layerVisibilityToggleButton',
        onClick: () => {
          this._closePopover();
          this.props.toggleVisible();
        }
      }
    ];

    if (!this.props.isReadOnly) {
      actionItems.push({
        name: i18n.translate('xpack.maps.layerTocActions.editLayerTitle', {
          defaultMessage: 'Edit layer',
        }),
        icon: (
          <EuiIcon
            type="pencil"
            size="m"
          />
        ),
        'data-test-subj': 'editLayerButton',
        onClick: () => {
          this._closePopover();
          this.props.editLayer();
        }
      });
      actionItems.push({
        name: i18n.translate('xpack.maps.layerTocActions.cloneLayerTitle', {
          defaultMessage: 'Clone layer',
        }),
        icon: (
          <EuiIcon
            type="copy"
            size="m"
          />
        ),
        'data-test-subj': 'cloneLayerButton',
        onClick: () => {
          this._closePopover();
          this.props.cloneLayer();
        }
      });
      actionItems.push({
        name: i18n.translate('xpack.maps.layerTocActions.removeLayerTitle', {
          defaultMessage: 'Remove layer',
        }),
        icon: (
          <EuiIcon
            type="trash"
            size="m"
          />
        ),
        'data-test-subj': 'removeLayerButton',
        onClick: () => {
          this._closePopover();
          this.props.removeLayer();
        }
      });
    }

    return {
      id: 0,
      title: i18n.translate('xpack.maps.layerTocActions.layerActionsTitle', {
        defaultMessage: 'Layer actions',
      }),
      items: actionItems,
    };
  }

  render() {
    return (
      <EuiPopover
        id="contextMenu"
        className="mapLayTocActions"
        button={this._renderPopoverToggleButton()}
        isOpen={this.state.isPopoverOpen}
        closePopover={this._closePopover}
        panelPaddingSize="none"
        withTitle
        anchorPosition="leftUp"
        anchorClassName="mapLayTocActions__popoverAnchor"
      >
        <EuiContextMenu
          initialPanelId={0}
          panels={[this._getActionsPanel()]}
          data-test-subj={`layerTocActionsPanel${this.props.escapedDisplayName}`}
        />
      </EuiPopover>);
  }
}
