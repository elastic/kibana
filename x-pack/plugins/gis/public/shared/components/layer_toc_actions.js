/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React, { Component } from 'react';

import {
  EuiButtonEmpty,
  EuiPopover,
  EuiContextMenu,
  EuiIcon,
  EuiLoadingSpinner,
  EuiToolTip,
  EuiIconTip
} from '@elastic/eui';


function flattenPanelTree(tree, array = []) {
  array.push(tree);

  if (tree.items) {
    tree.items.forEach(item => {
      if (item.panel) {
        flattenPanelTree(item.panel, array);
        item.panel = item.panel.id;
      }
    });
  }

  return array;
}

export class LayerTocActions extends Component {


  state = {
    isPopoverOpen: false
  };

  _onClick = () => {
    this.setState(prevState => ({
      isPopoverOpen: !prevState.isPopoverOpen,
    }));
  };

  _closePopover = () => {
    this.setState(() => ({
      isPopoverOpen: false
    }));
  };

  _renderButton() {
    const icon = this._renderIcon();
    return (
      <EuiButtonEmpty
        size="xs"
        flush="left"
        color="text"
        onClick={this._onClick}
      >
        {icon}
      </EuiButtonEmpty>);
  }


  _getVisbilityIcon() {
    const iconType = this.props.layer.isVisible() ? 'eye' : 'eyeClosed';
    return (
      <EuiIcon
        type={iconType}
        size="m"
      />);
  }

  _renderIcon() {
    const { zoom, layer } = this.props;
    let smallLegendIcon;
    if (layer.hasErrors()) {
      smallLegendIcon = (
        <EuiIconTip
          aria-label="Load warning"
          size="m"
          type="alert"
          color="warning"
          content={layer.getErrors()}
        />
      );
    } else if (layer.isLayerLoading()) {
      smallLegendIcon = <EuiLoadingSpinner size="m"/>;
    } else if (!layer.showAtZoomLevel(zoom)) {
      const { minZoom, maxZoom } = layer.getZoomConfig();
      const icon = layer.getIcon();
      smallLegendIcon = (
        <EuiToolTip
          position="top"
          content={`Map is at zoom level ${zoom}.
          This layer is only visible between zoom levels ${minZoom} to ${maxZoom}.`}
        >
          {icon}
        </EuiToolTip>
      );
    } else {
      smallLegendIcon = layer.getIcon();
    }
    return smallLegendIcon;
  }

  _getPanels() {

    const visibilityToggle = this._getVisbilityIcon();
    const panelTree = {
      id: 0,
      title: 'Layer actions',
      items: [
        {
          name: 'Fit to data',
          icon: (
            <EuiIcon
              type="search"
              size="m"
            />
          ),
          onClick: () => {
            this._closePopover();
            this.props.fitToBounds();
          },
        },
        {
          name: this.props.layer.isVisible() ? 'Hide layer' : 'Show layer',
          icon: visibilityToggle,
          onClick: () => {
            this._closePopover();
            this.props.toggleVisible();
          }
        }
      ],
    };

    return flattenPanelTree(panelTree);
  }

  render() {
    return (
      <EuiPopover
        id="contextMenu"
        button={this._renderButton()}
        isOpen={this.state.isPopoverOpen}
        closePopover={this._closePopover}
        panelPaddingSize="none"
        withTitle
        anchorPosition="leftUp"
      >
        <EuiContextMenu
          initialPanelId={0}
          panels={this._getPanels()}
        />
      </EuiPopover>);
  }
}
