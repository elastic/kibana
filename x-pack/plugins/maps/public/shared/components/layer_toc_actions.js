/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React, { Component } from 'react';

import {
  EuiButtonEmpty,
  EuiPopover,
  EuiContextMenu,
  EuiIcon,
  EuiLoadingSpinner,
  EuiToolTip,
  EuiIconTip
} from '@elastic/eui';
import { i18n } from '@kbn/i18n';

function flattenPanelTree(tree, array = []) {
  array.push(tree);

  if (tree.items) {
    tree.items.forEach(item => {
      if (item.panel) {
        flattenPanelTree(item.panel, array);
        item.panel = item.panel.id;
      }
    });
  }

  return array;
}

function cleanDisplayName(displayName) {
  if (!displayName) {
    return displayName;
  }
  return displayName.split(' ').join('');
}

export class LayerTocActions extends Component {

  state = {
    isPopoverOpen: false,
    supportsFitToBounds: false,
  };

  componentDidMount() {
    this._isMounted = true;
    this._loadSupportsFitToBounds();
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  async _loadSupportsFitToBounds() {
    const supportsFitToBounds = await this.props.layer.supportsFitToBounds();
    if (this._isMounted) {
      this.setState({ supportsFitToBounds });
    }
  }

  _onClick = () => {
    this.setState(prevState => ({
      isPopoverOpen: !prevState.isPopoverOpen,
    }));
  };

  _closePopover = () => {
    this.setState(() => ({
      isPopoverOpen: false
    }));
  };

  _renderButton() {
    const icon = this._renderIcon();
    return (
      <EuiButtonEmpty
        size="xs"
        flush="left"
        color="text"
        onClick={this._onClick}
        data-test-subj={`layerTocActionsPanelToggleButton${cleanDisplayName(this.props.displayName)}`}
      >
        {icon}
      </EuiButtonEmpty>);
  }


  _getVisbilityIcon() {
    const iconType = this.props.layer.isVisible() ? 'eye' : 'eyeClosed';
    return (
      <EuiIcon
        type={iconType}
        size="m"
      />);
  }

  _renderIcon() {
    const { zoom, layer } = this.props;
    let smallLegendIcon;
    if (layer.hasErrors()) {
      smallLegendIcon = (
        <EuiIconTip
          aria-label={i18n.translate('xpack.maps.layerTocActions.loadWarningAriaLabel', { defaultMessage: 'Load warning' })}
          size="m"
          type="alert"
          color="warning"
          content={layer.getErrors()}
        />
      );
    } else if (layer.isLayerLoading()) {
      smallLegendIcon = <EuiLoadingSpinner size="m"/>;
    } else if (!layer.showAtZoomLevel(zoom)) {
      const { minZoom, maxZoom } = layer.getZoomConfig();
      const icon = layer.getIcon();
      smallLegendIcon = (
        <EuiToolTip
          position="top"
          content={
            i18n.translate('xpack.maps.layerTocActions.zoomFeedbackTooltip', {
              defaultMessage: `Map is at zoom level {zoom}.
          This layer is only visible between zoom levels {minZoom} to {maxZoom}.`,
              values: { minZoom, maxZoom, zoom }
            })}
        >
          {icon}
        </EuiToolTip>
      );
    } else {
      smallLegendIcon = layer.getIcon();
    }
    return smallLegendIcon;
  }

  _getPanels() {

    const visibilityToggle = this._getVisbilityIcon();
    const panelTree = {
      id: 0,
      title: i18n.translate('xpack.maps.layerTocActions.layerActionsTitle', {
        defaultMessage: 'Layer actions',
      }),
      items: [
        {
          name: i18n.translate('xpack.maps.layerTocActions.fitToDataTitle', {
            defaultMessage: 'Fit to data',
          }),
          icon: (
            <EuiIcon
              type="search"
              size="m"
            />
          ),
          'data-test-subj': 'fitToBoundsButton',
          toolTipContent: this.state.supportsFitToBounds ? null : i18n.translate('xpack.maps.layerTocActions.noFitSupportTooltip', {
            defaultMessage: 'Layer does not support fit to data',
          }),
          disabled: !this.state.supportsFitToBounds,
          onClick: () => {
            this._closePopover();
            this.props.fitToBounds();
          },
        },
        {
          name: this.props.layer.isVisible() ? i18n.translate('xpack.maps.layerTocActions.hideLayerTitle', {
            defaultMessage: 'Hide layer',
          }) : i18n.translate('xpack.maps.layerTocActions.showLayerTitle', {
            defaultMessage: 'Show layer',
          }),
          icon: visibilityToggle,
          'data-test-subj': 'layerVisibilityToggleButton',
          onClick: () => {
            this._closePopover();
            this.props.toggleVisible();
          }
        }
      ],
    };

    return flattenPanelTree(panelTree);
  }

  render() {
    return (
      <EuiPopover
        id="contextMenu"
        button={this._renderButton()}
        isOpen={this.state.isPopoverOpen}
        closePopover={this._closePopover}
        panelPaddingSize="none"
        withTitle
        anchorPosition="leftUp"
      >
        <EuiContextMenu
          initialPanelId={0}
          panels={this._getPanels()}
          data-test-subj={`layerTocActionsPanel${cleanDisplayName(this.props.displayName)}`}
        />
      </EuiPopover>);
  }
}
