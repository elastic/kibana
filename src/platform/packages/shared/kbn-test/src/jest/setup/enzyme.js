/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

/* eslint-env jest */

/**
 * This is a workaround to fix snapshot serialization of emotion classes when rendering React@18 using `import { render } from 'enzyme'`
 * With React@18 emotion uses `useInsertionEffect` to insert styles into the DOM, which enzyme `render` does not trigger because it is using ReactDomServer.renderToString.
 * With React@17 emotion fell back to sync cb, so it was working with enzyme `render`.
 * Without those styles in DOM the custom snapshot serializer is not able to replace the emotion class names.
 * This workaround ensures a fake emotion style tag is present in the DOM before rendering the component with enzyme making the snapshot serializer work.
 */
function mockEnsureEmotionStyleTag() {
  if (!document.head.querySelector('style[data-emotion]')) {
    const style = document.createElement('style');
    style.setAttribute('data-emotion', 'css');
    document.head.appendChild(style);
  }
}

jest.mock('enzyme', () => {
  const actual = jest.requireActual('enzyme');
  return {
    ...actual,
    render: (node, options) => {
      mockEnsureEmotionStyleTag();
      return actual.render(node, options);
    },
  };
});
