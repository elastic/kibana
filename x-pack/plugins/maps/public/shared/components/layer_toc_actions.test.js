/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { LayerTocActions } from './layer_toc_actions';

let supportsFitToBounds;
let isLayerLoading;
let isVisible;
let hasErrors;
let showAtZoomLevel;
const layerMock = {
  supportsFitToBounds: () => { return supportsFitToBounds; },
  isVisible: () => { return isVisible; },
  hasErrors: () => { return hasErrors; },
  getErrors: () => { return 'simulated layer error'; },
  isLayerLoading: () => { return isLayerLoading; },
  showAtZoomLevel: () => { return showAtZoomLevel; },
  getZoomConfig: () => { return { minZoom: 2, maxZoom: 3 }; },
  getIcon: () => { return (<div>icon mock</div>); },
};

const defaultProps = {
  displayName: 'layer1',
  zoom: 0,
  layer: layerMock,
};

describe('LayerTocActions', () => {
  beforeEach(() => {
    supportsFitToBounds = true;
    isLayerLoading = false;
    isVisible = true;
    hasErrors = false;
    showAtZoomLevel = true;
  });

  test('is rendered', async () => {
    const component = shallow(
      <LayerTocActions
        {...defaultProps}
      />
    );

    // Ensure all promises resolve
    await new Promise(resolve => process.nextTick(resolve));
    // Ensure the state changes are reflected
    component.update();

    expect(component)
      .toMatchSnapshot();
  });

  test('should disable fit to data when supportsFitToBounds is false', async () => {
    supportsFitToBounds = false;
    const component = shallow(
      <LayerTocActions
        {...defaultProps}
      />
    );

    // Ensure all promises resolve
    await new Promise(resolve => process.nextTick(resolve));
    // Ensure the state changes are reflected
    component.update();

    expect(component)
      .toMatchSnapshot();
  });

  test('should display spinner when layer is loading', async () => {
    isLayerLoading = true;
    const component = shallow(
      <LayerTocActions
        {...defaultProps}
      />
    );

    // Ensure all promises resolve
    await new Promise(resolve => process.nextTick(resolve));
    // Ensure the state changes are reflected
    component.update();

    expect(component)
      .toMatchSnapshot();
  });

  test('should show warning when layer has errors', async () => {
    hasErrors = true;
    const component = shallow(
      <LayerTocActions
        {...defaultProps}
      />
    );

    // Ensure all promises resolve
    await new Promise(resolve => process.nextTick(resolve));
    // Ensure the state changes are reflected
    component.update();

    expect(component)
      .toMatchSnapshot();
  });

  test('should show visible toggle when layer is not visible', async () => {
    isVisible = false;
    const component = shallow(
      <LayerTocActions
        {...defaultProps}
      />
    );

    // Ensure all promises resolve
    await new Promise(resolve => process.nextTick(resolve));
    // Ensure the state changes are reflected
    component.update();

    expect(component)
      .toMatchSnapshot();
  });

  test('should provide feedback when layer is not visible because of current zoom level', async () => {
    showAtZoomLevel = false;
    const component = shallow(
      <LayerTocActions
        {...defaultProps}
      />
    );

    // Ensure all promises resolve
    await new Promise(resolve => process.nextTick(resolve));
    // Ensure the state changes are reflected
    component.update();

    expect(component)
      .toMatchSnapshot();
  });
});
