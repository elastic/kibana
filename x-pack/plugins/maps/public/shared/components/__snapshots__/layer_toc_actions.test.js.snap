// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayerTocActions is rendered 1`] = `
<EuiPopover
  anchorPosition="leftUp"
  button={
    <EuiButtonEmpty
      color="text"
      data-test-subj="layerTocActionsPanelToggleButtonlayer1"
      flush="left"
      iconSide="left"
      onClick={[Function]}
      size="xs"
      type="button"
    >
      <div>
        icon mock
      </div>
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  hasArrow={true}
  id="contextMenu"
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="none"
  withTitle={true}
>
  <EuiContextMenu
    data-test-subj="layerTocActionsPanellayer1"
    initialPanelId={0}
    panels={
      Array [
        Object {
          "id": 0,
          "items": Array [
            Object {
              "data-test-subj": "fitToBoundsButton",
              "disabled": false,
              "icon": <EuiIcon
                size="m"
                type="search"
              />,
              "name": "Fit to data",
              "onClick": [Function],
              "toolTipContent": null,
            },
            Object {
              "data-test-subj": "layerVisibilityToggleButton",
              "icon": <EuiIcon
                size="m"
                type="eye"
              />,
              "name": "Hide layer",
              "onClick": [Function],
            },
            Object {
              "data-test-subj": "cloneLayerButton",
              "icon": <EuiIcon
                size="m"
                type="copy"
              />,
              "name": "Clone layer",
              "onClick": [Function],
            },
          ],
          "title": "Layer actions",
        },
      ]
    }
  />
</EuiPopover>
`;

exports[`LayerTocActions should disable fit to data when supportsFitToBounds is false 1`] = `
<EuiPopover
  anchorPosition="leftUp"
  button={
    <EuiButtonEmpty
      color="text"
      data-test-subj="layerTocActionsPanelToggleButtonlayer1"
      flush="left"
      iconSide="left"
      onClick={[Function]}
      size="xs"
      type="button"
    >
      <div>
        icon mock
      </div>
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  hasArrow={true}
  id="contextMenu"
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="none"
  withTitle={true}
>
  <EuiContextMenu
    data-test-subj="layerTocActionsPanellayer1"
    initialPanelId={0}
    panels={
      Array [
        Object {
          "id": 0,
          "items": Array [
            Object {
              "data-test-subj": "fitToBoundsButton",
              "disabled": true,
              "icon": <EuiIcon
                size="m"
                type="search"
              />,
              "name": "Fit to data",
              "onClick": [Function],
              "toolTipContent": "Layer does not support fit to data",
            },
            Object {
              "data-test-subj": "layerVisibilityToggleButton",
              "icon": <EuiIcon
                size="m"
                type="eye"
              />,
              "name": "Hide layer",
              "onClick": [Function],
            },
            Object {
              "data-test-subj": "cloneLayerButton",
              "icon": <EuiIcon
                size="m"
                type="copy"
              />,
              "name": "Clone layer",
              "onClick": [Function],
            },
          ],
          "title": "Layer actions",
        },
      ]
    }
  />
</EuiPopover>
`;

exports[`LayerTocActions should display spinner when layer is loading 1`] = `
<EuiPopover
  anchorPosition="leftUp"
  button={
    <EuiButtonEmpty
      color="text"
      data-test-subj="layerTocActionsPanelToggleButtonlayer1"
      flush="left"
      iconSide="left"
      onClick={[Function]}
      size="xs"
      type="button"
    >
      <EuiLoadingSpinner
        size="m"
      />
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  hasArrow={true}
  id="contextMenu"
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="none"
  withTitle={true}
>
  <EuiContextMenu
    data-test-subj="layerTocActionsPanellayer1"
    initialPanelId={0}
    panels={
      Array [
        Object {
          "id": 0,
          "items": Array [
            Object {
              "data-test-subj": "fitToBoundsButton",
              "disabled": false,
              "icon": <EuiIcon
                size="m"
                type="search"
              />,
              "name": "Fit to data",
              "onClick": [Function],
              "toolTipContent": null,
            },
            Object {
              "data-test-subj": "layerVisibilityToggleButton",
              "icon": <EuiIcon
                size="m"
                type="eye"
              />,
              "name": "Hide layer",
              "onClick": [Function],
            },
            Object {
              "data-test-subj": "cloneLayerButton",
              "icon": <EuiIcon
                size="m"
                type="copy"
              />,
              "name": "Clone layer",
              "onClick": [Function],
            },
          ],
          "title": "Layer actions",
        },
      ]
    }
  />
</EuiPopover>
`;

exports[`LayerTocActions should not show edit actions in read only mode 1`] = `
<EuiPopover
  anchorPosition="leftUp"
  button={
    <EuiButtonEmpty
      color="text"
      data-test-subj="layerTocActionsPanelToggleButtonlayer1"
      flush="left"
      iconSide="left"
      onClick={[Function]}
      size="xs"
      type="button"
    >
      <div>
        icon mock
      </div>
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  hasArrow={true}
  id="contextMenu"
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="none"
  withTitle={true}
>
  <EuiContextMenu
    data-test-subj="layerTocActionsPanellayer1"
    initialPanelId={0}
    panels={
      Array [
        Object {
          "id": 0,
          "items": Array [
            Object {
              "data-test-subj": "fitToBoundsButton",
              "disabled": false,
              "icon": <EuiIcon
                size="m"
                type="search"
              />,
              "name": "Fit to data",
              "onClick": [Function],
              "toolTipContent": null,
            },
            Object {
              "data-test-subj": "layerVisibilityToggleButton",
              "icon": <EuiIcon
                size="m"
                type="eye"
              />,
              "name": "Hide layer",
              "onClick": [Function],
            },
          ],
          "title": "Layer actions",
        },
      ]
    }
  />
</EuiPopover>
`;

exports[`LayerTocActions should provide feedback when layer is not visible because of current zoom level 1`] = `
<EuiPopover
  anchorPosition="leftUp"
  button={
    <EuiButtonEmpty
      color="text"
      data-test-subj="layerTocActionsPanelToggleButtonlayer1"
      flush="left"
      iconSide="left"
      onClick={[Function]}
      size="xs"
      type="button"
    >
      <EuiToolTip
        content="Map is at zoom level 0.
          This layer is only visible between zoom levels 2 to 3."
        delay="regular"
        position="top"
      >
        <div>
          icon mock
        </div>
      </EuiToolTip>
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  hasArrow={true}
  id="contextMenu"
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="none"
  withTitle={true}
>
  <EuiContextMenu
    data-test-subj="layerTocActionsPanellayer1"
    initialPanelId={0}
    panels={
      Array [
        Object {
          "id": 0,
          "items": Array [
            Object {
              "data-test-subj": "fitToBoundsButton",
              "disabled": false,
              "icon": <EuiIcon
                size="m"
                type="search"
              />,
              "name": "Fit to data",
              "onClick": [Function],
              "toolTipContent": null,
            },
            Object {
              "data-test-subj": "layerVisibilityToggleButton",
              "icon": <EuiIcon
                size="m"
                type="eye"
              />,
              "name": "Hide layer",
              "onClick": [Function],
            },
            Object {
              "data-test-subj": "cloneLayerButton",
              "icon": <EuiIcon
                size="m"
                type="copy"
              />,
              "name": "Clone layer",
              "onClick": [Function],
            },
          ],
          "title": "Layer actions",
        },
      ]
    }
  />
</EuiPopover>
`;

exports[`LayerTocActions should show visible toggle when layer is not visible 1`] = `
<EuiPopover
  anchorPosition="leftUp"
  button={
    <EuiButtonEmpty
      color="text"
      data-test-subj="layerTocActionsPanelToggleButtonlayer1"
      flush="left"
      iconSide="left"
      onClick={[Function]}
      size="xs"
      type="button"
    >
      <div>
        icon mock
      </div>
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  hasArrow={true}
  id="contextMenu"
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="none"
  withTitle={true}
>
  <EuiContextMenu
    data-test-subj="layerTocActionsPanellayer1"
    initialPanelId={0}
    panels={
      Array [
        Object {
          "id": 0,
          "items": Array [
            Object {
              "data-test-subj": "fitToBoundsButton",
              "disabled": false,
              "icon": <EuiIcon
                size="m"
                type="search"
              />,
              "name": "Fit to data",
              "onClick": [Function],
              "toolTipContent": null,
            },
            Object {
              "data-test-subj": "layerVisibilityToggleButton",
              "icon": <EuiIcon
                size="m"
                type="eyeClosed"
              />,
              "name": "Show layer",
              "onClick": [Function],
            },
            Object {
              "data-test-subj": "cloneLayerButton",
              "icon": <EuiIcon
                size="m"
                type="copy"
              />,
              "name": "Clone layer",
              "onClick": [Function],
            },
          ],
          "title": "Layer actions",
        },
      ]
    }
  />
</EuiPopover>
`;

exports[`LayerTocActions should show warning when layer has errors 1`] = `
<EuiPopover
  anchorPosition="leftUp"
  button={
    <EuiButtonEmpty
      color="text"
      data-test-subj="layerTocActionsPanelToggleButtonlayer1"
      flush="left"
      iconSide="left"
      onClick={[Function]}
      size="xs"
      type="button"
    >
      <EuiIconTip
        aria-label="Load warning"
        color="warning"
        content="simulated layer error"
        size="m"
        type="alert"
      />
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  hasArrow={true}
  id="contextMenu"
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="none"
  withTitle={true}
>
  <EuiContextMenu
    data-test-subj="layerTocActionsPanellayer1"
    initialPanelId={0}
    panels={
      Array [
        Object {
          "id": 0,
          "items": Array [
            Object {
              "data-test-subj": "fitToBoundsButton",
              "disabled": false,
              "icon": <EuiIcon
                size="m"
                type="search"
              />,
              "name": "Fit to data",
              "onClick": [Function],
              "toolTipContent": null,
            },
            Object {
              "data-test-subj": "layerVisibilityToggleButton",
              "icon": <EuiIcon
                size="m"
                type="eye"
              />,
              "name": "Hide layer",
              "onClick": [Function],
            },
            Object {
              "data-test-subj": "cloneLayerButton",
              "icon": <EuiIcon
                size="m"
                type="copy"
              />,
              "name": "Clone layer",
              "onClick": [Function],
            },
          ],
          "title": "Layer actions",
        },
      ]
    }
  />
</EuiPopover>
`;
